### PR TITLE
Unify zig/C environ

### PR DESCRIPTION
**Pull request by marler8997**

### CODE REVIEW

File: lib/std/os.zig [hunk 1]
```diff
@@ -50,8 +50,25 @@ pub const system = if (builtin.link_libc) std.c else switch (builtin.os) {
 
 pub usingnamespace @import("os/bits.zig");
 
-/// See also `getenv`. Populated by startup code before main().
-pub var environ: [][*]u8 = undefined;
+// Using this as I don't know how to declare a different version of environ
```
Comment (daurnimator) [inline lib/std/os.zig]:
```zig
comptime {
    if (builtin.link_libc) {
        ......
    } else {
        ......
    }
}
```

Comment (LemonBoy) [reply lib/std/os.zig]: AFAIR you can't put declarations in `comptime`, I usually use something like this:

```zig
pub usingnamespace if (builtin.link_libc) struct {
//
} else {
//
};
```

**Comment by daurnimator** [comment]
This will need rebasing on top of the sentinel pointer merge.

**Comment by andrewrk** [comment]
This has been a draft for over a month, please re-open when it's ready for review.

**Comment by marler8997** [comment]
@andrewrk I was just pushing change to this today.  It looks like when it's closed the testing it no longer enabled and I'm unable to reopen it (I may not have permission).  Should I create a new PR?

**Comment by marler8997** [comment]
Sorry it's been open so long, I can't reproduce the issue occuring on the builder and I'm not sure why.  I've come back to it a few times but still can't seem to figure out why I'm getting the `lld: error: unable to find library -lC` error.

**Comment by emekoi** [comment]
perhaps this is because libc is special-cased in the compiler?

**Comment by daurnimator** [comment]
@emekoi it was because of incorrect case to `extern "C"` (should be `extern "c"` instead).

**Comment by andrewrk** [comment]
please re-open when you have something ready for review

**Pull request closed without merging**